src/usermeta.ts

```diff
@@ -31,20 +32,78 @@ export interface UsermetaClientOptions {
 }
 
+export type KeyValue = { key: string, value: string };
+
 export interface UsermetaClientBulkOptions extends UsermetaClientOptions {
   usernames: string[];
 };
 
 export type UsermetaClientCallback = (err: Error | null, reply?: string | null) => void;
-export type UsermetaClientBulkCallback = (err: Error | null, reply?: object[] | null) => void;
+export type UsermetaClientBulkCallback = (err: Error | null, reply?: object[] | null | string) => void;
 
 export interface SimpleUsermetaClient {
-  set: (params:UsermetaClientOptions|string, key:string, value:string, callback:UsermetaClientCallback, maxLength?:number) => void;
-  get: (params:UsermetaClientOptions|string, key:string, callback:UsermetaClientCallback) => void;
+  set: (params: UsermetaClientOptions | string, key: string, value: string, callback: UsermetaClientCallback, maxLength?: number) => void;
+  get: (params: UsermetaClientOptions | string, key: string, callback: UsermetaClientCallback) => void;
+  getBulk: (pparams: UsermetaClientBulkOptions | string, keys: string[], cb: UsermetaClientBulkCallback) => void;
+  setBulk: (pparams: UsermetaClientBulkOptions | string, keyValues: KeyValue[], cb: UsermetaClientBulkCallback) => void;
 };
 
-export interface BulkedUsermetaClient extends SimpleUsermetaClient {
-  getBulk: (params: UsermetaClientBulkOptions | string, key: string[], callback: UsermetaClientBulkCallback) => void;
+export class BulkedUsermetaClient {
+  getBulk(pparams: UsermetaClientBulkOptions | string, keys: string[], cb: UsermetaClientBulkCallback) {
+    let tasks: any[] = [];
+    let userNames: string[] = [];
+    if (typeof pparams === 'object' && (pparams as UsermetaClientBulkOptions).usernames.length > 0) {
+      userNames = (pparams as UsermetaClientBulkOptions).usernames;
+    } else {
+      userNames = [(pparams as string)];
+    }
+    userNames.forEach((username) => {
+      let clonedParams: any;
+      if (typeof pparams === 'object') {
+        clonedParams = Object.assign({}, pparams);
+        clonedParams.username = username;
+      } else
+        clonedParams = username.toString();
+
+      keys.forEach((key) => {
+
+        tasks.push(cb2 => this['get'](clonedParams, key, (err: Error | null, reply?: string | null) => {
+
+          if (err) {
+            return cb2(err, { username, key, value: '' });
+          }
+          cb2(err, { username, key, value: reply });
+        }))
+      });
+    });
+
+    async.parallel(tasks, (err, data) => {
+      if (err) {
+        cb(err, []);
+      }
+      else {
+        //make the 2 levels array to 1 level => [[], [], []] => []
+        let oneLevelArray = (data as any[][])?.flat();
+        cb(null, oneLevelArray);
+      }
+    });
+  }
+  setBulk(pparams: UsermetaClientBulkOptions | string, keyValues: KeyValue[], cb: UsermetaClientBulkCallback) {
+    let tasks: any[] = [];
+    keyValues.forEach((kv) => {
+      tasks.push(cb2 => this['set'](pparams, kv.key, kv.value, cb2))
+    });
```

Comment:
For good style, don't write this:

```ts
let out = [];
in.forEach(x => out.push(...));
```

But write this:

```ts
const out = in.map(x => ...);
```

In the code above:

```ts
const tasks:Callback[] = keyValues.map((kv) => (cb2 => this['set'](pparams, kv.key, kv.value, cb2)));
```